themes/wilson_theme_starterkit/src/js/components/main-menu.js

```diff
@@ -13,102 +13,13 @@
    */
   Drupal.behaviors.mainMenu = {
     attach(context) {
-      // Only progress if there is a menu element to work with.
-      if (context.querySelector(".navigation nav .menu")) {
-        const body = context.querySelector("body");
-        const nav = context.querySelector(".navigation nav");
-        const topLevelMenu = context.querySelector(".navigation .menu");
-        const topLevelSubmenuLinks = context.querySelectorAll(
-          ".navigation .menu > li.has-submenu > a, .navigation .menu > li.has-submenu > span"
-        );
-        const submenuLinks = context.querySelectorAll(
-          ".navigation li.has-submenu > a, .navigation li.has-submenu > span"
-        );
-        const backLinks = context.querySelectorAll(".back-link a");
+      const body = document.querySelector("body");
+      const nav = context.querySelector(".navigation nav");
```

Comment:
I'm totally not suggeseting you make changes to this, but I wonder if this is the kind of thing that we could make customisable in the theme settings - the navigation class, I mean... right now we'd have to keep the templates/classes/structure the same for the menu to work, right?

Which isn't so bad. But maybe on another site we'd want to call it (I dunno) `.doohickey` ... or they don't have a navigation wrapper, maybe it's just `header > nav` or `nav.main-navigation` etc. 

These could be set in the theme settings (Appearance > Settings or wherever) and default to `.navigation nav` and then those could be added to the `DrupalSettings` object.

Anyway. Going back to reviewing the actual code rather than making silly suggestions :D

Reply:
That's an interesting idea, and not something I'd thought about. The template, JS and styling for this menu is already overly complex and interwoven, so the added complexity of customisable selectors worries me a little (particularly with how that ties in with compiled CSS). I'd also say that the pattern of configuring selectors/classes in the theme settings/config isn't immediately familiar, so maybe a bit of cognitive overhead there.

Wilson's main menu is very opinionated because they're a pain to build custom each time. The idea was that you get a pretty sophisticated mobile and desktop menu out of the box, and that the designers should design to that, to cut dev effort. In hindsight, that probably was a utopian ideal and heavy customisation will always be necessary. Eugh

There's probably more than can be done to simplify what we have here (like removing `>` selectors), so that changing the structure slightly doesn't immediately cause problems. I could also change the `.navigation` class to be something that seems more important and relevant to what it does (e.g. `.responsive-nav`). That way, if you do remove the class, you can feel a bit foolish when the responsive menu stops working 😄 

I'll have another look in to this around some other work.

Reply:
> particularly with how that ties in with compiled CSS

Yeah, that makes sense, though I suppose a lot of folks would be updating/overriding elements of the navigation styling for a custom theme.

I didn't mean to give you a migraine :joy: it was just a thought that I had and shared without thinking about it first! It would be interesting to see if this is something that could be done, but definitely not anything that needs to be done.

Reply:
I think I agree with Paul that customisable selectors probably wouldn't be worth the extra complexity for the benefits.

Not that I do a huge amount of theming but if I was changing the structure of the HTML for something I'd expect to have to update any JS that works on that HTML. Maybe we could add comments to templates that add JS-required classes saying which JS depends on them instead?

Or maybe change how the required classes are added so that they're added in preprocess functions (or other non-template files) and as long as the overriding template added `{{ classes }}` or `{{ attributes }}` they'd still exist? That might be a hefty undertaking to update across the whole theme though.

(Both would be future, follow-up tasks rather than part of this though)